.github/workflows/cut-new-release.yml

```diff
@@ -0,0 +1,100 @@
+name: Cut a new release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        type: string
+        description: Version of the next release (e.g. v0.15.0)
+        required: true
+
+env:
+  git-user: github-actions[bot]
+  git-email: 41898282+github-actions[bot]@users.noreply.github.com
+  main-branch: main
+
+jobs:
+  workflow-metadata:
+    name: Extract workflow metadata
+    runs-on: ubuntu-22.04
+    outputs:
+      ref: ${{ steps.extract.outputs.ref }}
+      base-version: ${{ steps.extract.outputs.base-version }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.GH_TOKEN }}
```

Comment:
why we need to use a token here? IMO it's not needed here as we don't commit/push here, please remove if so.

Reply:
I think removing this breaks the workflow trigger, will test